### ISSUE TITLE
read can not have position excceed buffer length 

### ISSUE THREAD

**Issue by dding3** [dmlc/xgboost#5863]
Tried to train a XGBRegressor with XGBoost4J-Spark as:
`val xgbRf0 = new XGBoostRegressor()
xgbRf0.setNumRound(500)
    xgbRf0.setMaxDepth(50)
    xgbRf0.setNthread(20)
    xgbRf0.setTreeMethod("hist")
    xgbRf0.setSeed(2)
    xgbRf0.setEta(0.1)
    xgbRf0.setMinChildWeight(1)
    xgbRf0.setSubsample(0.8)
    xgbRf0.setColsampleBytree(0.8)
    xgbRf0.setGamma(0)
    xgbRf0.setAlpha(0)
    xgbRf0.setLambda(1)
    xgbRf0.setNumWorkers(3)`

The training dataset has more than 6 million records, during training, I have below error with xgboost 0.90:
_20/06/23 01:15:21 INFO java.RabitTracker$TrackerProcessLogger: 2020-06-23 01:15:21,250 INFO [497]       train-rmse:0.006187
20/06/23 01:15:24 INFO java.RabitTracker$TrackerProcessLogger: 2020-06-23 01:15:24,192 INFO [498]       train-rmse:0.006187
20/06/23 01:15:28 INFO java.RabitTracker$TrackerProcessLogger: 2020-06-23 01:15:28,023 INFO [499]       train-rmse:0.006188
20/06/23 01:15:28 INFO scheduler.DAGScheduler: Job 27 finished: foreachPartition at XGBoost.scala:452, took 1852.658991 s
20/06/23 01:15:28 INFO spark.SparkContext: Starting job: first at XGBoost.scala:570
**AssertError:read can not have position excceed buffer length**_

And I have core dump on executor with xgboost master.

Tried both java serializer and kryo serializer, no luck.

**Comment by thvasilo** [comment]
Most likely similar to this: https://github.com/dmlc/xgboost/issues/5650

**Comment by trivialfis** [comment]
This seems to be a rabit error.  It can happen if something went wrong during synchronization between workers.

**Comment by dding3** [comment]
@thvasilo Thank you for your reply. 

Seems for [#5650](https://github.com/dmlc/xgboost/issues/5650), the exception happened when loading a newer model with old version. I trained the model from scratch and the exception happened during training. Any suggestion to walk around the exception?

**Comment by trivialfis** [comment]
I don't think it's related to model loading.

**Comment by thvasilo** [comment]
In that case this is an error I think I've seen with rabbit trying to write more bytes to the sync buffer because it does read-ahead in chunks of 1024 bytes. See this issue for example: https://github.com/apache/incubator-datasketches-cpp/issues/5

**Comment by trivialfis** [comment]
Could you please try latest xgboost?
